dspy/adapters/chat_adapter.py

```diff
@@ -168,11 +169,19 @@ def parse(self, signature: Type[Signature], completion: str) -> dict[str, Any]:
                 try:
                     fields[k] = parse_value(v, signature.output_fields[k].annotation)
                 except Exception as e:
-                    raise ValueError(
-                        f"Error parsing field {k}: {e}.\n\n\t\tOn attempting to parse the value\n```\n{v}\n```"
+                    raise AdapterParseError(
+                        adapter_name="ChatAdapter",
+                        signature=signature,
+                        lm_response=completion,
+                        message=f"Failed to parse field {k} with value {v} from the LM response. Error message: {e}",
                     )
         if fields.keys() != signature.output_fields.keys():
-            raise ValueError(f"Expected {signature.output_fields.keys()} but got {fields.keys()}")
+            raise AdapterParseError(
+                adapter_name="ChatAdapter",
+                signature=signature,
+                lm_response=completion,
+                parsed_result=fields,
```

Comment:
What happened to the error message?

Reply:
It's generated inside the `AdapterParseError` dynamically based on the attributes. Sometimes we don't have the partial parsing, and sometimes we want to provide a custom message.